packages/react-dom-bindings/src/client/ReactDOMFloatClient.js

```diff
@@ -865,135 +906,141 @@ export function hydrateHoistable(
+      case 'defaultChecked':
+      case 'innerHTML':
+      case 'suppressContentEditableWarning':
+      case 'suppressHydrationWarning':
+      case 'style':
+        // we advance to the next prop
         continue;
+
+      // Name remapped props used by hoistable tag types
+      case 'className': {
+        if (__DEV__) {
+          checkAttributeStringCoercion(propValue, propName);
+        }
+        attributes.push('class', '' + propValue);
+        break;
       }
-      switch (propName) {
-        // Reserved props will never have an attribute partner
-        case 'children':
-        case 'defaultValue':
-        case 'dangerouslySetInnerHTML':
-        case 'defaultChecked':
-        case 'innerHTML':
-        case 'suppressContentEditableWarning':
-        case 'suppressHydrationWarning':
-        case 'style':
-          // we advance to the next prop
-          continue;
-
-        // Name remapped props used by hoistable tag types
-        case 'className': {
-          if (__DEV__) {
-            checkAttributeStringCoercion(propValue, propName);
-          }
-          if (node.getAttribute('class') !== '' + propValue) continue nodeLoop;
-          break;
+      case 'httpEquiv': {
+        if (__DEV__) {
+          checkAttributeStringCoercion(propValue, propName);
         }
-        case 'httpEquiv': {
-          if (__DEV__) {
-            checkAttributeStringCoercion(propValue, propName);
-          }
-          if (node.getAttribute('http-equiv') !== '' + propValue)
-            continue nodeLoop;
-          break;
+        attributes.push('http-equiv', '' + propValue);
+        break;
+      }
+
+      // Boolean props used by hoistable tag types
+      case 'async':
+      case 'defer':
+      case 'disabled':
+      case 'hidden':
+      case 'noModule':
+      case 'scoped':
+      case 'itemScope':
+        if (propValue === true) {
+          attributes.push(propName, '');
         }
+        break;
 
-        // Booleanish props used by hoistable tag types
-        case 'contentEditable':
-        case 'draggable':
-        case 'spellCheck': {
+      // The following properties are left out because they do not apply to
+      // the current set of hoistable types. They may have special handling
+      // requirements if they end up applying to a hoistable type in the future
+      // case 'acceptCharset':
+      // case 'value':
+      // case 'allowFullScreen':
+      // case 'autoFocus':
+      // case 'autoPlay':
+      // case 'controls':
+      // case 'default':
+      // case 'disablePictureInPicture':
+      // case 'disableRemotePlayback':
+      // case 'formNoValidate':
+      // case 'loop':
+      // case 'noValidate':
+      // case 'open':
+      // case 'playsInline':
+      // case 'readOnly':
+      // case 'required':
+      // case 'reversed':
+      // case 'seamless':
+      // case 'multiple':
+      // case 'selected':
+      // case 'capture':
+      // case 'download':
```

Comment:
On the server these two get treated as overloaded boolean so a false means no attribute. practically speaking no one is going to use these on hoistable tag types so feels a shame to include them however if one were to use them then hydration would break.

Currently I'm ok with that edge case but wanted to point it out for possible challenge